pkg/updater/service/setup_updater.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// Package service provides a way to interact with os services
+package service
+
+import "github.com/DataDog/datadog-agent/pkg/util/log"
+
+const updaterUnit = "datadog-updater.service"
+
+// todo: make this independent of agent bootstrap
+func setupUpdater() (err error) {
+	defer func() {
+		if err != nil {
+			log.Errorf("Failed to setup updater: %s", err)
+			RemoveUpdater()
+		}
+	}()
+	if err = loadUnit(updaterUnit); err != nil {
+		return
+	}
+	if err = systemdReload(); err != nil {
+		return
+	}
+	if err = enableUnit(updaterUnit); err != nil {
+		return
+	}
+	if err = startUnit(updaterUnit); err != nil {
+		return
+	}
+	return nil
+}
+
+// RemoveUpdater stops and removes updater units
+func RemoveUpdater() {
```

Comment:
missing the call on the Purge command, will send the fix tomorrow as github ignores my pushes